uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/controllers/sub-controllers/toil-request-ctrl_test.js

```diff
@@ -225,9 +230,64 @@
           it('does show balance', function () {
             expect($ctrl.uiOptions.showBalance).toBeTruthy();
           });
+
+          describe('expiry date', function () {
+            beforeEach(function () {
+              $ctrl.calculateToilExpiryDate();
+            });
+
+            it('is not obtained from server', function () {
+              expect(AbsenceTypeAPI.calculateToilExpiryDate).not.toHaveBeenCalled();
+            });
+          });
         });
       });
 
+      describe('when manager opens leave request popup', function () {
+        beforeEach(function () {
+          var status = optionGroupMock.specificValue('hrleaveandabsences_leave_request_status', 'value', '3');
+          var toilRequest = TOILRequestInstance.init(mockData.findBy('status_id', status));
+          toilRequest.contact_id = CRM.vars.leaveAndAbsences.contactId.toString();
+          var directiveOptions = {
+            contactId: 203, //manager's contact id
+            leaveRequest: toilRequest
+          };
+
+          initTestController(directiveOptions);
+        });
+
+        it('manager role is set', function () {
+          expect($ctrl.isRole('manager')).toBeTruthy();
+        });
+
+        it('expiry date is set on ui', function () {
+          expect($ctrl.uiOptions.expiryDate).toEqual(jasmine.any(Date));
+        });
+
+        describe('and changes expiry date', function () {
+          var beforeExpiryDate;
+
+          beforeEach(function () {
+            beforeExpiryDate = $ctrl.request.toil_expiry_date;
+            $ctrl.uiOptions.expiryDate = new Date();
+            $ctrl.updateExpiryDate();
+          });
+
+          it('sets new expiry date', function () {
+            expect(beforeExpiryDate).not.toEqual($ctrl.request.toil_expiry_date);
```

Comment:
also, please test that the new value of `$ctrl.request.toil_expiry_date` is what you expect